OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -409,6 +420,53 @@ else if (action.getUrlTarget() == OSInAppMessageAction.OSInAppMessageActionUrlTy
         }
     }
 
+    private void fireRESTCallForPageChange(@NonNull final OSInAppMessage message, @NonNull final OSInAppMessagePage page) {
+        final String variantId = variantIdForMessage(message);
+        if (variantId == null)
+            return;
+
+        final String pageId = page.getPageId();
+
+
+        // Never send multiple page impressions for the same message UUID unless that page change is from an IAM with redisplay
+        if (message.getViewedPageIds().contains(pageId))
```

Comment:
this is checked under onPageChanged too, is that ok?

Reply:
Now only checking in the rest method call for consistency with iOS and better separation of concerns